### PR TITLE
Fix Embedding layer to check invalid inputs

### PR DESCRIPTION
**Pull request by yanboliang**
Bug: If users set ```input_length``` for ```Embedding``` layer incorrectly, it should throw exception(existing not). Otherwise, the downstream layers compute shape incorrect and throw confused exception.
I have fixed this at https://github.com/keras-team/keras/pull/11091, but it's different from ```tf.keras```. ```keras-team/keras``` makes this check in ```compute_output_shape``` which was executed always. But ```tf.keras``` only calls ```compute_output_shape``` in deferred mode, so we need to move this check to other place. Actually it makes more sense to move it to ```_assert_input_compatibility``` which is used to check the inputs legal, and would be run in all mode.

### CODE REVIEW

File: tensorflow/python/keras/layers/embeddings_test.py [hunk 1]
```diff
@@ -92,5 +92,23 @@ def test_eager_gpu_cpu(self):
     opt.apply_gradients(zip(gs, l.weights))
     self.assertAllEqual(len(gs), 1)
 
+  @tf_test_util.run_in_graph_and_eager_modes(use_gpu=False)
+  def test_embedding_invalid(self):
+    # len(input_length) should be equal to len(input_shape) - 1
+    with self.assertRaises(ValueError):
```
Comment (fchollet) [inline tensorflow/python/keras/layers/embeddings_test.py]: Please check that the expected error message in being raised (otherwise the test could pass due to an unrelated exception).

File: tensorflow/python/keras/layers/embeddings_test.py [hunk 2]
```diff
@@ -92,5 +92,23 @@ def test_eager_gpu_cpu(self):
...
+      model = keras.Sequential([keras.layers.Embedding(
+          input_dim=10,
+          output_dim=4,
+          input_length=2,
+          input_shape=(3, 4, 5))])
+
+    # input_length should be equal to input_shape[1:]
+    with self.assertRaises(ValueError):
```
Comment (fchollet) [inline tensorflow/python/keras/layers/embeddings_test.py]: Please check that the expected error message in being raised (otherwise the test could pass due to an unrelated exception).

**Review by fchollet — COMMENTED** [review]
Thank you for your PR. It looks good to me.

**Comment by yanboliang** [comment]
@fchollet I have updated the PR according your review. Thanks.

**Comment by yanboliang** [comment]
Just resolved the conflicts.

**Comment by qlzh727** [comment]
Ping @fchollet for review.

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @fchollet: You have been added as a reviewer to this pull request. Please add your review or reassign. It has been 14 days with no activity and the `awaiting review` label has been applied.

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @fchollet: You have been added as a reviewer to this pull request. Please add your review or reassign. It has been 29 days with no activity and the `awaiting review` label has been applied.

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @fchollet: You have been added as a reviewer to this pull request. Please add your review or reassign. It has been 44 days with no activity and the `awaiting review` label has been applied.

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @fchollet: You have been added as a reviewer to this pull request. Please add your review or reassign. It has been 59 days with no activity and the `awaiting review` label has been applied.
